src/test/java/com/microsoft/azure/relay/HybridConnectionListenerTest.java

```diff
@@ -289,28 +290,46 @@ public void parseRequestPathAndQueryTest() throws IOException, URISyntaxExceptio
 	public void listenerReconnectionTest() {
 		AtomicInteger handlerExecuted = new AtomicInteger(0);
 		AtomicReference<Throwable> exception = new AtomicReference<>(new Exception());
-		
-		listener.setConnectingHandler((ex) -> {
+		listener.setConnectingHandler(ex -> {
 			exception.set(ex);
 			handlerExecuted.incrementAndGet();
 		});
 		
 		listener.openAsync(Duration.ofSeconds(15)).join();
-		ControlConnection controlConnection = listener.getControlConnection();
-		controlConnection.getConnectAsyncTask().thenCompose(controlWebSocket -> {
-			return controlWebSocket.closeAsync(); // This bypasses listener.close(), simulating an unexpected close
-		}).join();
-		
+		listener.injectFault(new ConnectionLostException());
+		assertFalse("listener should be disconnected temporarily for now", listener.isOnline());
```

Comment:
>assertFalse [](start = 2, length = 11)

Is there ever a race between injectFault causing the websocket to get disposed and this line? I think if the `controlConnection.connectAsyncTask` is already completed then when it calls `.thenApply` to dispose the websocket that might happen immediately? (or will that run it async?)

Reply:
Theoretically there is most definitely a race. However, disposing the websocket is a VERY LONG operation (takes 10 sec or more), I didn't want to be left hanging for that long, and isOnline() will return false when the call is made, not necessarily after the call finishes.

---
In reply to: [354528043](https://github.com/Azure/azure-relay-java/pull/58#discussion_r354528043) [](ancestors = 354528043)

Reply:
Bottom line is, I feel that as long as the websocket is closing and will get disposed, I don't think the user cares about how long it will take.

---
In reply to: [354529498](https://github.com/Azure/azure-relay-java/pull/58#discussion_r354529498) [](ancestors = 354529498,354528043)